sdk/modelsrepository/Azure.Iot.ModelsRepository/samples/ModelsRepositoryClientSamples/ModelResolutionSamples.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Azure.Iot.ModelsRepository.Samples
+{
+    internal class ModelResolutionSamples
+    {
+        private static string ClientSamplesDirectoryPath => Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+        private static string ClientSamplesLocalModelsRepository => Path.Combine(ClientSamplesDirectoryPath, "SampleModelsRepo");
+
+        public static void ClientInitialization()
+        {
+            // When no Uri is provided for instantiation. The global Azure IoT Models Repository endpoint is used
+            // and the dependency model resolution option is set to TryFromExpanded.
+            var globalRepoClient = new ModelsRepositoryClient();
+            Console.WriteLine($"Initialized client pointing to global endpoint: {globalRepoClient.RepositoryUri}");
+
+            // This form shows specifing a custom Uri for the models repository with default client options.
+            // The default client options will enable model dependency resolution.
+            var remoteRepoEndpoint = "https://contoso.com/models";
+            var remoteRepoClient = new ModelsRepositoryClient(new Uri(remoteRepoEndpoint));
+            Console.WriteLine($"Initialized client pointing to custom endpoint: {remoteRepoClient.RepositoryUri}");
+
+            // The client will also work with a local filesystem Uri. This example shows initalization
+            // with a local Uri and disabling model dependency resolution.
+            var localRepoPath = ClientSamplesLocalModelsRepository;
+            var localRepoClient = new ModelsRepositoryClient(
+                new Uri(localRepoPath),
+                new ModelsRepositoryClientOptions(resolutionOption: DependencyResolutionOption.Disabled));
+            Console.WriteLine($"Initialized client pointing to local path: {localRepoClient.RepositoryUri}");
+        }
+
+        public static async Task ResolveExistingModelsFromEndpointAsync()
```

Comment:
Now that we are changing the API name to GetModels, I would rename these methods too.
also `FromEndpoint` doesn't tell me much, maybe `FromGlobalRepo` or `FromDefaultRepo` ?

Reply:
Made the `FromEndpoint` -> `FromGlobalRepo` change as you suggested. But if you don't mind, will take care of the `Resolve()` -> `GetModels()` change in next PR.